tasks/connectors/edgescan/lib/edgescan_asset.rb

```diff
@@ -33,34 +34,34 @@ def location_specifiers
           end
         end
 
-        def find_location_specifier(specifier_id, location)
-          location_specifiers.find { |specifier| specifier.id == specifier_id } ||
-            location_specifiers.find { |specifier| specifier.location == location }
-        end
-
         # Converts an Edgescan asset into Kenna friendly ones
         #
-        # Edgescan and Kenna assets don't map one to one. A Kenna asset is more like an Edgescan
-        # location specifier. Because of that, one Edgescan asset usually gets turned into multiple
-        # Kenna assets.
-        #
         # This will:
-        # - Create Kenna assets based on Edgescan location specifiers
-        # - Go through the vulnerabilites and if some of them don't have a matching Edgescan
-        #   location specifier create Kenna assets for them
+        # - Create Kenna assets based on Edgescan hosts
+        # - If a matching host is not found an asset will be created from the vulnerability data
         def to_kenna_assets
-          location_specifiers_as_kenna_assets +
-            vulnerabilities_without_location_specifiers_as_kenna_assets
+          vulnerabilities.each_with_object([]) do |vulnerability, assets|
+            host = find_host(vulnerability.location)
+            asset = host ? host.to_corresponding_kenna_asset : vulnerability.to_corresponding_kenna_asset
+            assets << asset
+          end
         end
 
         private
 
-        def location_specifiers_as_kenna_assets
-          location_specifiers.map(&:to_kenna_asset)
+        def find_host(location)
+          host = hosts.find { |h| h.location == location }
+          return host unless host.nil?
+
+          host = hosts.find { |h| h.hostnames.each { |hostname| location.include?(hostname) } }
+          return host unless host.nil?
+
+          nil
         end
 
-        def vulnerabilities_without_location_specifiers_as_kenna_assets
-          vulnerabilities.reject(&:matching_location_specifier).map(&:to_corresponding_kenna_asset)
+        def find_location_specifier(specifier_id, location)
```

Comment:
IMO this code probably isn't needed as we should always be able to find a host in Edgescan if there's a vulnerability related to it - right?

Reply:
For you, yes you should always be able to find a Host since you're running ASM scans, for other clients running VM scans then a Host won't be available. 

Although, we're not using location specifiers anymore anyway so, removed in f69b7fc.